python/lance/types.py

```diff
@@ -0,0 +1,122 @@
+#  Copyright 2022 Lance Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Arrow extension types for Lance
+"""
+from abc import ABC, abstractproperty
+import platform
+
+import pandas as pd
+
+import pyarrow as pa
+
+
+__all__ = ["ImageType", "ImageUriType", "ImageBinaryType", "Point2dType", "Box2dType"]
+
+
+# Arrow extension type
+from pyarrow import ArrowKeyError
+
+
+class LanceType(pa.ExtensionType, ABC):
+    def __init__(self, storage_type, extension_name):
+        if platform.system() != "Linux":
+            raise NotImplementedError(
+                "Extension types are enabled for linux only for now"
+            )
+        super(LanceType, self).__init__(storage_type, extension_name)
+
+
+class ImageType(LanceType):
+    @abstractproperty
+    def storage_type(self):
+        pass
+
+    def __arrow_ext_serialize__(self):
+        return b""
```

Comment:
Curious what does this function do?

Reply:
For parametrized types this returns extra information (from the examples in Arrow docs)